File: php/src/Model/CreatedAttachment.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * The attachment record plus where to PUT its bytes.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class CreatedAttachment
{
    public function __construct(
        public readonly Attachment $attachment,
        public readonly AttachmentUpload $upload,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            attachment: Attachment::fromArray($data['attachment'] ?? []),
            upload: AttachmentUpload::fromArray($data['upload'] ?? []),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/CreatedAttachment.java
package io.fivexer.sdk.model;

/**
 * The attachment record plus where to PUT its bytes.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class CreatedAttachment {
    private Attachment attachment;
    private AttachmentUpload upload;

    public Attachment getAttachment() { return attachment; }
    public AttachmentUpload getUpload() { return upload; }
}
